src/sentry/auth/providers/fly/provider.py

```diff
@@ -1,15 +1,122 @@
+        return [
+            FlyOAuth2Login(client_id=self.get_client_id()),
+            OAuth2Callback(
+                access_token_url=ACCESS_TOKEN_URL,
+                client_id=self.get_client_id(),
+                client_secret=self.get_client_secret(),
+            ),
+            FetchUser(org=self.org),
+            # ConfirmEmail(),  # reference github provider
+        ]
+
+    def get_refresh_token_url(self):
+        return ACCESS_TOKEN_URL
+
+    def build_config(self, state: Any, organization: Optional[Any] = None):
+        """
+        On configuration, we determine which provider organization to configure SSO for
+        This configuration is then stored and passed into the pipeline instances during SSO
+        to determine whether the Auth'd user has the appropriate access to the provider org
+
+        {
+            'state': '9da4041848844e8088864eaea3c3a705',
+            'data': {
+                'access_token': 'fo1_6xgeCrB8ew8vFQ86vdaakBSFTVDGCzOUvebUbvgPGhI',
+                'token_type': 'Bearer',
+                'expires_in': 7200,
+                'refresh_token': 'PmUkAB75UPLKGZplERMq8WwOHnsTllZ5HveY4RvNUTk',
+                'scope': 'read',
+                'created_at': 1686786353},
+                'user': {
+                    'resource_owner_id': 'k9d01lp82rky6vo2',
+                    'scope': ['read'],
+                    'expires_in': 7200,
+                    'application': {'uid': 'elMJpuhA5bXbR59ZaKdXrxXGFVKTypGHuJ4h6Rfw1Qk'},
+                    'created_at': 1686786353,
+                    'user_id': 'k9d01lp82rky6vo2',
+                    'user_name': 'Nathan',
+                    'email': 'k9d01lp82rky6vo2@customer.fly.io',
+                    'organizations': [
+                        {'id': 'nathans-org', 'role': 'member'},
+                        {'id': '0vogzmzoj1k5xp29', 'role': 'admin'}
+                    ]
+                }
+            }
+        }
+        """
+        org = organization
+        if not organization:
+            data = state["data"]
+            # TODO: determine which org to configure SSO for
+            org = data["user"]["organizations"][0]
+
+        return {"org": org}
 
     def build_identity(self, state):
-        return {"id": "", "email": ""}
+        """
+        ex Response:
+        {
+            'resource_owner_id': 'k9d01lp82rky6vo2',
+            'scope': ['read'],
+            'expires_in': 7200,
+            'application': {
+                'uid': 'elMJpuhA5bXbR59ZaKdXrxXGFVKTypGHuJ4h6Rfw1Qk'
+            },
+            'created_at': 1686785304,
+            'user_id': 'k9d01lp82rky6vo2',
+            'user_name': 'Nathan',
+            'email': 'k9d01lp82rky6vo2@customer.fly.io',
+            'organizations': [
+                {'id': 'g1lx9my4pzemqwk7', 'role': 'admin'},
+                {'id': '0vogzmzoj1k5xp29', 'role': 'admin'}
+            ]
+        }
+        """
+        data = state["data"]
```

Comment:
nit: Would this be cleaner as `user_data = state["data"]["user"]`? Just opinion based

Reply:
so - we utilize `data` below in the return as an arg passed into `get_oauth_data`

but generally yes, we don't _need_ to break it down into multiple variables 🤷 
I prefer not to have the deep drilling of multiple `['foo']['bar']['biz']['etc.']` getters though. IMO makes it much more prone for issues